pkg/datapath/linux/config/config_test.go

```diff
@@ -51,6 +51,7 @@ func setup(tb testing.TB) {
 
 	require.NoError(tb, rlimit.RemoveMemlock(), "Failed to remove memory limits")
 
+	option.Config.EnableHostLegacyRouting = true // Disable obtaining direct routing device.
```

Comment:
Can this be moved closer to where it's used? Does this need to be set to true for all tests in ConfigSuite?

Reply:
Hey @ti-mo sorry for the long wait on this, was out for the past couple weeks. I believe this needs to be set for practically every test, which was what I did initially before this approach. Let me know what you think is better here, but from what I saw, nearly all tests needed it to pass.

Reply:
Marking resolved since this got approved, @ti-mo feel free to reopen if there is anything you'd like to see done here.